common/lib/xmodule/xmodule/x_module.py

```diff
@@ -881,7 +881,22 @@ def __getattr__(self, name):
                 request_post[key] = map(FileObjForWebobFiles, request.POST.getall(key))
 
         response_data = self.handle_ajax(suffix, request_post)
-        return Response(response_data, content_type='application/json')
+
+        try:
+            return Response(response_data, content_type='application/json')
+
+        except TypeError:
+            request_environ = getattr(request, 'environ')
+            log.exception(
+                "Type Error in xmodule handler WebOb response for request url: %s, response_data type: %s, LANG: %s, "
+                "LC_ALL: %s and HTTP_ACCEPT_ENCODING: %s",
+                request_environ['HTTP_REFERER'],
+                type(response_data),
+                request_environ['LANG'],
```

Comment:
As we are getting elements from `request_environ`, are we sure that every `request_environ` object will have all the keys (`LC_ALL` for example)  we are expecting?